src/cc65/pragma.c

```diff
@@ -430,13 +433,35 @@ static void SegNamePragma (StrBuf* B, segment_t Seg)
         goto ExitPoint;
     }
 
-    /* Get the string */
+    /* Get the name string of the segment */
     Name = SB_GetConstBuf (&S);
 
     /* Check if the name is valid */
     if (ValidSegName (Name)) {
 
-        /* Set the new name */
+        /* Skip the following comma */
+        SB_SkipWhite (B);
+        if (SB_Peek (B) == ',') {
+            SB_Skip (B);
+            SB_SkipWhite (B);
+
+            /* A string argument must follow */
+            if (!GetString (B, &A)) {
+                goto ExitPoint;
+            }
+
+            /* Get the address size string for the segment */
+            AddrSize = AddrSizeFromStr (SB_GetConstBuf (&A));
```

Comment:
That expression actually gets the size, not the "size string".  The `GetString()` call, above, got the string.